File: examples/functions/rest-api-methods/products/[id]/put.py
import json

def handler(event, id):
    """PUT /products/:id — id arrives directly from [id] filename"""
    body = event.get("body", "")
    try:
        data = json.loads(body) if isinstance(body, str) else (body or {})
    except Exception:
        return {"status": 400, "body": {"error": "Invalid JSON"}}

    return {
        "status": 200,
        "body": {"id": int(id), **data, "updated": True},
    }

File: examples/functions/rest-api-methods/products/[id]/put.lua
-- PUT /products/:id — id arrives directly from [id] filename
local cjson = require("cjson")

local function handler(event, params)
    local id = params.id or ""
    local ok, data = pcall(cjson.decode, event.body or "")
    if not ok then
        return { status = 400, body = cjson.encode({ error = "Invalid JSON" }) }
    end

    data.id = tonumber(id)
    data.updated = true
    return {
        status = 200,
        headers = { ["Content-Type"] = "application/json" },
        body = cjson.encode(data),
    }
end

return handler
